std/path.d

```diff
@@ -3956,23 +3956,31 @@ string expandTilde(string inputPath) nothrow
         }
 
         // Replaces the tilde from path with the environment variable HOME.
-        static string expandFromEnvironment(string path) nothrow
+        static string expandFromEnvironment(string path) nothrow @safe
         {
             import core.stdc.stdlib : getenv;
+            import core.stdc.string : strlen;
+            import std.exception : assumeUnique;
 
             assert(path.length >= 1);
             assert(path[0] == '~');
 
             // Get HOME and use that to replace the tilde.
-            auto home = getenv("HOME");
-            if (home == null)
-                return path;
+            auto home = () @trusted {
+                auto homeStr = getenv("HOME");
+                if (homeStr == null)
+                    return path;
+                return homeStr[0 .. strlen(homeStr)].assumeUnique;
+            } ();
+
+            if (home == path)
+                return home;
 
             return combineCPathWithDPath(home, path, 1);
         }
 
         // Replaces the tilde from path with the path from the user database.
-        static string expandFromDatabase(string path) nothrow
+        static string expandFromDatabase(string path) nothrow @trusted
```

Comment:
Hmmm that's quite a huge `@trusted` - can't we use a ScopedAllocator from stdx.allocator? This should make all the manual allocations safe.

Reply:
`ScopedAllocator`, even with `GCAllocator` is also `@system`

Reply:
I agree that this is far too much code to mark as `@trusted`.

I wonder if it's really necessary to use `malloc` here. How about using a static array buffer instead, and fall back to a GC-allocated array in the few cases where the username is uncommonly long?

Reply:
Ah, I only saw the `malloc` and didn't notice the `realloc` calls before now. But I actually think it's OK to use plain GC allocations for both `username` and `extra_memory`.